builtin/k8s/platform.go

```diff
@@ -1309,6 +1343,30 @@ type AutoscaleConfig struct {
 type Pod struct {
 	SecurityContext *PodSecurityContext `hcl:"security_context,block"`
 	Container       *Container          `hcl:"container,block"`
+	Sidecars        []*SidecarContainer `hcl:"sidecar,block"`
+}
+
+// SidecarContainer describes the configuration for the sidecar container
+type SidecarContainer struct {
```

Comment:
Hm, rather than having a `SidecarContainer` struct, we should just move all of these over to the existing `Container` struct. I don't see anything specific to sidecars that requires it to be separate.